source/common/router/scoped_rds.cc

```diff
@@ -349,18 +349,19 @@ void ScopedRdsConfigSubscription::onConfigUpdate(
   onConfigUpdate(to_add_repeated, to_remove_repeated, version_info);
 }
 
-std::string ScopedRdsConfigSubscription::loadTypeUrl() {
-  switch (xds_api_version_) {
+std::string ScopedRdsConfigSubscription::loadTypeUrl(
+    envoy::config::core::v3alpha::ApiVersion resource_api_version) {
+  switch (resource_api_version) {
   // automatically set api version as V2
-  case envoy::config::core::v3alpha::ConfigSource::AUTO:
-  case envoy::config::core::v3alpha::ConfigSource::V2:
+  case envoy::config::core::v3alpha::ApiVersion::AUTO:
+  case envoy::config::core::v3alpha::ApiVersion::V2:
```

Comment:
non-immediately actionable comment: is it worth it to sit down with the protobuf team at some point and show them everything we are doing? IMO this is a very unique and interesting case, and I do feel like with some better/different built-in descriptor iteration support we could avoid a huge amount of boilerplate like this. Meaning, it could all be done via annotations and then a map would be loaded at runtime. I know I've commented on this before but it seems worth tracking and/or having a discussion with them.

Reply:
Yeah, we should probably do that once the dust settles on v3, there's a bunch of places we've had to do gymnastics. I'll take a note to followup, thanks.